services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -0,0 +1,212 @@
+            def dpId = (disc.rule =~ /sw:(.*?),/)[0][1]
+            def cookie = (disc.rule =~ /ck:(.*?),/)[0][1]
+            cookies[dpId] = cookie
+        }
+        return cookies
+    }
+
+    boolean findDirectLinks(Switch src, Switch dst, List<IslInfoData> links) {
+        def connectingLinks = links.findAll {link ->
+            (link.path[0].switchId == src.dpId) && (link.path[-1].switchId == dst.dpId)
+        }
+        return connectingLinks.empty
+    }
+
+    def getNonNeighbouringSwitches() {
+        def islInfoData = northbound.getAllLinks()
+        def switches = topologyDefinition.getActiveSwitches()
+        def differentSwitches = [switches, switches].combinations().findAll {src, dst -> src.dpId != dst.dpId}
+
+        return differentSwitches.find {src, dst -> findDirectLinks(src, dst, islInfoData)}
+    }
+
+    def getNeighbouringSwitches() {
+        def switches = topologyDefinition.getActiveSwitches()
+        def islInfoData = northbound.getAllLinks()
+        return [switches, switches].combinations().unique().find { src, dst -> !findDirectLinks(src, dst, islInfoData) }
+    }
+
+    def getSingleSwitch() {
+        def switches = topologyDefinition.getActiveSwitches()
+        return [switches.first(), switches.first()]
+    }
+
+    def deleteFlow(String flowId) {
+        try {
+            northbound.deleteFlow(flowId)
+        } catch (e) {
+            log.error("Unable to delete flow " + flowId, e)
+        }
+    }
+
+    @Unroll
+    def "Flow validation should fail if #rule switch rule is deleted"() {
+        given: "Flow with a transit switch exists"
+        def (Switch srcSwitch, Switch dstSwitch) = getNonNeighbouringSwitches()
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow = northbound.addFlow(flow)
+
+        assert Wrappers.wait(20) { "up".equalsIgnoreCase(northbound.getFlow(flow.id).status) }
+        def forward = database.getFlow(flow.id)?.left
+        def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id)*.dpId
+
+        and: "Rules are installed on all switches"
+        assert Wrappers.wait(30, 2) { involvedSwitches.every { sw ->
+            northbound.getSwitchRules(sw).flowEntries.find { it.cookie == forward.cookie }
+        }
+        }
+        when: "${rule} switch rule is deleted"
+        def dpId = involvedSwitches[item]
+        def cookie = forward.cookie
+        northbound.deleteSwitchRules(dpId, cookie)
+
+        def validationResult = northbound.validateFlow(flow.id)
+        deleteFlow(flow?.id)
+
+        then: "Forward and reverse flows should be validated"
+        validationResult.size() == 2
+
+        and: "Reverse flow validation must be successful"
+        def reverseResult = validationResult.find { isFlowValid(it) }
+        !reverseResult.toString().contains(cookie.toString())
+
+        and: "Forward flow validation should fail"
+        def forwardResult = validationResult.find { !isFlowValid(it) }
```

Comment:
This block itself is not performing verification. Note that `find` method will return `null` if no results found (will not throw or sth), so you should add
```
and: "Forward flow validation should fail"
def forwardResult = validationResult.find { !isFlowValid(it) }
forwardResult
```
to verify that forwardResult is actually found. Otherwise you may just assign 'null' to forwardResult and continue test execution.  
Same in the above block, but at least it will throw a NPE at line 113 (which is still not the best way to fail the test, though still viable)